internal/store/certificatesigningrequest.go

```diff
@@ -267,3 +268,38 @@ func (s *CertificateSigningRequestStore) UpdateConditions(ctx context.Context, o
 		return s.updateConditions(orgId, name, conditions)
 	})
 }
+
+func (s *CertificateSigningRequestStore) updateCertificate(orgId uuid.UUID, name string, certificate []byte) (bool, error) {
+	existingRecord := model.CertificateSigningRequest{Resource: model.Resource{OrgID: orgId, Name: name}}
+	result := s.db.First(&existingRecord)
+	if result.Error != nil {
+		return false, flterrors.ErrorFromGormError(result.Error)
+	}
+
+	if existingRecord.Status == nil {
+		existingRecord.Status = model.MakeJSONField(api.CertificateSigningRequestStatus{})
+	}
+	existingRecord.Status.Data.Certificate = &certificate
+
+	// TODO - this currently updates the resource version *in addition to* the update that occurs when the resource's Condition
+	// is set to approved. these are technically two distinct operations but in reality will generally bundled together. should we
+	// only update the resource version once between them?
+	result = s.db.Model(existingRecord).Where("resource_version = ?", lo.FromPtr(existingRecord.ResourceVersion)).Updates(map[string]interface{}{
+		"status":           existingRecord.Status,
+		"resource_version": gorm.Expr("resource_version + 1"),
```

Comment:
I wonder about resource version, generally i believe status should not update resource version so why it should be done here?

Reply:
Sorry actually my fault here, it should be updated though this is more k8s functionality for etcd and not sure why it is needed here

Reply:
I am following the example of enrollmentrequests's [`UpdateStatus`](https://github.com/flightctl/flightctl/blob/main/internal/store/enrollmentrequest.go#L204).